src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/FunctionDescr.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Reflection;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    internal class FunctionDescr
+    {
+        internal string Name { get; }
+
+        internal MethodInfo Method { get; }
+
+        internal FormulaType RetType { get; }
+
+        // User-facing parameter types. 
+        internal FormulaType[] ParamTypes { get; }
+
+        // If not null, then arg0 is from RuntimeConfig
+        internal Type ConfigType { get; }
+
+        internal bool IsAsync { get; }
+
+        internal BigInteger LamdaParamMask { get; }
+
+        internal IEnumerable<CustomFunctionSignatureHelper> ArgumentSignatures { get; }
```

Comment:
>internal IEnumerable<CustomFunctionSignatureHelper> ArgumentSignatures { get; } [](http://example.com/codeflow?start=8&length=79)

new field.
Can possibly be null. but that's taken care in CustomTexlFunction and CustomSetPropertyFunction's ctor